crates/ruma-common/src/events/room/message/audio.rs

```diff
@@ -26,16 +26,30 @@ pub struct AudioMessageEventContent {
 }
 
 impl AudioMessageEventContent {
+    /// Creates a new `AudioMessageEventContent` with the given body and source.
+    pub fn new(body: String, source: MediaSource) -> Self {
+        Self { body, source, info: None }
+    }
+
     /// Creates a new non-encrypted `AudioMessageEventContent` with the given body, url and
     /// optional extra info.
-    pub fn plain(body: String, url: OwnedMxcUri, info: Option<Box<AudioInfo>>) -> Self {
-        Self { body, source: MediaSource::Plain(url), info }
+    pub fn plain(body: String, url: OwnedMxcUri) -> Self {
+        Self::new(body, MediaSource::Plain(url))
     }
 
     /// Creates a new encrypted `AudioMessageEventContent` with the given body and encrypted
     /// file.
     pub fn encrypted(body: String, file: EncryptedFile) -> Self {
-        Self { body, source: MediaSource::Encrypted(Box::new(file)), info: None }
+        Self::new(body, MediaSource::Encrypted(Box::new(file)))
+    }
+
+    /// Creates a new `AudioMessageEventContent` from `self` with the `info` field set to the given
+    /// value.
+    ///
+    /// Since the field is public, you can also assign to it directly. This method merely acts
+    /// as a shorthand for that, because it is very common to set this field.
+    pub fn info(self, info: impl Into<Option<Box<AudioInfo>>>) -> Self {
```

Comment:
Couldn't we just have `Into<Box<AudioInfo>>` here? I'm guessing this is going to be used like a builder pattern and by default it is `None`. And same with the other builder-like methods in the other types?

Reply:
We could, but conditionally calling that method based on whether an `Option` you have is `Some` or `None` is annoying.

Reply:
Then I guess that's Ok